### PR TITLE
[TE] Set up ThirdEye swagger UI and add API documentations

### PR DESCRIPTION
**Pull request by jihaozh**
- Set up swagger UI to allow user to easily visualize and interact with ThirdEye API's resources

- Add swagger documentations for some endpoints

### CODE REVIEW

File: thirdeye/thirdeye-pinot/pom.xml [hunk 1]
```diff
@@ -231,6 +231,13 @@
       <version>1.4.192</version>
       <scope>test</scope>
     </dependency>
+
+    <!-- Swagger -->
+    <dependency>
+      <groupId>io.federecio</groupId>
+      <artifactId>dropwizard-swagger</artifactId>
+      <version>0.7.0</version>
```
Comment (cyenjung) [inline thirdeye/thirdeye-pinot/pom.xml]: Declare the version number at the parent level, i.e., thirdeye/pom.xml.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/ThirdEyeAnomalyApplication.java [hunk 2]
```diff
@@ -70,6 +73,16 @@ public String getName() {
   @Override
   public void initialize(final Bootstrap<ThirdEyeAnomalyConfiguration> bootstrap) {
     bootstrap.addBundle(new AssetsBundle("/assets/", "/", "index.html"));
+    bootstrap.addBundle(new SwaggerBundle<ThirdEyeConfiguration>() {
```
Comment (cyenjung) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/ThirdEyeAnomalyApplication.java]: Extract the class of swaggerbundle and reuse it in both anomaly and dashboard applications.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/AnomalyResource.java [hunk 3]
```diff
@@ -138,17 +120,29 @@ public AnomalyResource(AnomalyFunctionFactory anomalyFunctionFactory, AlertFilte
...
-  public List<MergedAnomalyResultDTO> viewMergedAnomaliesInRange(@NotNull @QueryParam("dataset") String dataset,
-      @QueryParam("startTimeIso") String startTimeIso, @QueryParam("endTimeIso") String endTimeIso,
-      @QueryParam("metric") String metric, @QueryParam("dimensions") String exploredDimensions,
+  @ApiOperation(value = "View merged anomalies for collection")
+  public List<MergedAnomalyResultDTO> viewMergedAnomaliesInRange(
+      @ApiParam(value = "dataset", required = true)
+      @NotNull @QueryParam("dataset") String dataset,
+      @ApiParam(value = "startTimeIso")
+      @QueryParam("startTimeIso") String startTimeIso,
+      @ApiParam(value = "endTimeIso")
```
Comment (cyenjung) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/AnomalyResource.java]: Is it end time exclusive or inclusive? Please refine this description and other descriptions of ApiParam.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/AnomalyResource.java [hunk 4]
```diff
@@ -833,7 +830,11 @@ private void toggleFunctionById(long id, boolean isActive) {
   // Delete anomaly function
   @DELETE
   @Path("/anomaly-function")
-  public Response deleteAnomalyFunctions(@NotNull @QueryParam("id") Long id,
+  @ApiOperation(value = "Delete anomaly function")
+  public Response deleteAnomalyFunctions(
+      @ApiParam(value = "id")
```
Comment (cyenjung) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/AnomalyResource.java]: anomaly function id

**Comment by kishoreg** [comment]
This is awesome!. Do we have a documentation on API's?

**Comment by jihaozh** [comment]
Thank you. I've written documentation to some of theAPIs. With swagger, we can write more later.

**Pull request merged**